core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -427,23 +435,21 @@ private[ui] class JobDataSource(
     val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
     val submissionTime = jobData.submissionTime
     val formattedSubmissionTime = submissionTime.map(UIUtils.formatDate).getOrElse("Unknown")
-    val lastStageAttempt = store.lastStageAttempt(jobData.stageIds.max)
-    val lastStageDescription = lastStageAttempt.description.getOrElse("")
+    val (lastStageName, lastStageDescription) = lastStageNameAndDescription(store, jobData)
 
-    val formattedJobDescription =
-      UIUtils.makeDescription(lastStageDescription, basePath, plainText = false)
+    val jobDescription = UIUtils.makeDescription(lastStageDescription, basePath, plainText = false)
```

Comment:
No need to check for empty description here?

Reply:
`lastStageDescription` may be empty, but it will not cause problems, `makeDescription` will handle it properly, just like in the version before lastStageAttempt was used:
```    
  val jobDescription = UIUtils.makeDescription(jobData.description.getOrElse(""), 
  basePath, plainText = false)
```

Reply:
Sure, but don't you want the same behavior as above here (falling back to the job name)?

Reply:
I've moved this logic to `lastStageNameAndDescription`, so it's uniform.